packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1299,3 +1290,25 @@ class DebugConnectionInfo {
   final Uri wsUri;
   final String baseUri;
 }
+
+/// Returns the next platform value for the switcher.
+///
+/// These values must match what is available in
+/// packages/flutter/lib/src/foundation/binding.dart
+String nextPlatform(String currentPlatform, FeatureFlags featureFlags) {
+  switch (currentPlatform) {
+    case 'android':
+      return 'iOS';
+    case 'iOS':
+      return 'fuchsia';
+    case 'fuchsia':
+      if (featureFlags.isMacOSEnabled) {
+        return 'macOS';
+      }
+      return 'android';
+    case 'macOS':
+      return 'android';
+    default:
```

Comment:
What does it mean `currentPlatform` was something else? Should we emit a warning here?

Reply:
Perhaps there was a bug in the serialization code on the framework side, or a bug else where. Maybe we could assert here?

Reply:
Updated to use an assert

Reply:
LGTM